sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClient.java

```diff
@@ -253,29 +287,15 @@ private Channel addMetadata(Channel channel, String tenantId) {
         channel, MetadataUtils.newAttachHeadersInterceptor(metadata));
   }
 
-  private String encodeKeyspacePrefix(String tenantId) {
-    return Hex.encodeHexString(tenantId.getBytes(StandardCharsets.UTF_8)) + "_";
-  }
-
-  private String addTenantPrefix(String keyspaceName) {
-    return tenantPrefix + keyspaceName;
-  }
-
-  private CqlKeyspaceDescribe stripTenantPrefix(CqlKeyspaceDescribe describe) {
-    if (describe == null || tenantPrefix.isEmpty()) {
-      return describe;
-    }
-    CqlKeyspace keyspace = describe.getCqlKeyspace();
-    return CqlKeyspaceDescribe.newBuilder(describe)
-        .setCqlKeyspace(
-            CqlKeyspace.newBuilder(keyspace).setName(stripTenantPrefix(keyspace.getName())).build())
-        .build();
-  }
-
-  private String stripTenantPrefix(String fullKeyspaceName) {
-    if (fullKeyspaceName.startsWith(tenantPrefix)) {
-      return fullKeyspaceName.substring(tenantPrefix.length());
+  static class DescribeKeyspaceObserver extends UnaryStreamObserver<CqlKeyspaceDescribe> {
+    @Override
+    public void onError(Throwable t) {
+      if (t instanceof StatusRuntimeException
+          && ((StatusRuntimeException) t).getStatus().getCode() == Status.Code.NOT_FOUND) {
+        onNext(null);
+      } else {
+        super.onError(t);
+      }
```

Comment:
ok so this is a hack.. not sure if this should not be directly handled in the handler, this way is a bit hidden.. Leaving to you..